### PR TITLE
Add test for deployer_envsubst_base.

### PR DESCRIPTION
**Pull request by trironkk**

### CODE REVIEW

File: testing.Makefile [hunk 1]
```diff
@@ -52,4 +52,25 @@ testing/marketplace/deployer/helm_tiller_onbuild: \
 		.testing/marketplace/deployer/helm_tiller_onbuild/standard
 
 
+.testing/marketplace/deployer/envsubst:
+	mkdir -p $@
+
+.testing/marketplace/deployer/envsubst/standard: \
+    .build/marketplace/deployer/envsubst \
```
Comment (huyhg) [inline testing.Makefile]: we can follow the convention in this file for how these lines are indented? I think the problem is tabs and spaces are being mixed

File: testing.Makefile [hunk 2]
```diff
@@ -52,4 +52,25 @@ testing/marketplace/deployer/helm_tiller_onbuild: \
...
+		.build/var/MARKETPLACE_TOOLS_TAG \
+		.build/var/REGISTRY \
+		$(shell find testing/marketplace/deployer_envsubst/standard -type f) \
+		testing.Makefile \
+    | .testing/marketplace/deployer/envsubst
+	$(call print_target)
+	TEST_ID=$(TEST_ID) \
+	REGISTRY=$(REGISTRY) \
+	MARKETPLACE_TOOLS_TAG=$(MARKETPLACE_TOOLS_TAG) \
+    ./testing/marketplace/deployer_envsubst_base/standard/test
```
Comment (huyhg) [inline testing.Makefile]: needs tabs

File: testing/marketplace/deployer_envsubst_base/standard/test [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (huyhg) [inline testing/marketplace/deployer_envsubst_base/standard/test]: can we call this file `run_test`?

File: testing/marketplace/deployer_envsubst_base/standard/test [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+set -eox pipefail
+
+test_directory=$(mktemp --dry-run)
+cp -r $(dirname "$0") "$test_directory"
+cd "$test_directory"
+
+image="$REGISTRY/testing/deployer_envsubst/1:$MARKETPLACE_TOOLS_TAG-$TEST_ID"
```
Comment (huyhg) [inline testing/marketplace/deployer_envsubst_base/standard/test]: s/1/standard/. Or some bash construct to extract the last portion of the directory path

File: testing/marketplace/deployer_envsubst_base/standard/test [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+docker push "$image"
+
+docker run \
+    "gcr.io/cloud-marketplace-tools/k8s/dev:$MARKETPLACE_TOOLS_TAG" \
+    cat /scripts/dev \
+  > dev
+chmod a+x dev
+
+export MARKETPLACE_TOOLS_IMAGE="gcr.io/cloud-marketplace-tools/k8s/dev"
```
Comment (huyhg) [inline testing/marketplace/deployer_envsubst_base/standard/test]: need export the TAG (instead)

File: testing/marketplace/deployer_envsubst_base/standard/test [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+app_parameters="{}"
+./dev \
+  /scripts/verify \
+      --deployer="$image" \
+      --parameters="$app_parameters"
+
```
Comment (huyhg) [inline testing/marketplace/deployer_envsubst_base/standard/test]: rm

File: testing/marketplace/deployer_envsubst_base/standard/test [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  > dev
+chmod a+x dev
+
+export MARKETPLACE_TOOLS_IMAGE="gcr.io/cloud-marketplace-tools/k8s/dev"
+
+app_parameters="{}"
+./dev \
+  /scripts/verify \
+      --deployer="$image" \
+      --parameters="$app_parameters"
```
Comment (huyhg) [inline testing/marketplace/deployer_envsubst_base/standard/test]: use single quotes--we were hit by this before

File: testing.Makefile (hunk 1 above)
Comment (trironkk) [reply testing.Makefile]: Fixed.

File: testing.Makefile (hunk 2 above)
Comment (trironkk) [reply testing.Makefile]: Done.

File: testing/marketplace/deployer_envsubst_base/standard/test (hunk 3 above)
Comment (trironkk) [reply testing/marketplace/deployer_envsubst_base/standard/test]: Sure!

File: testing/marketplace/deployer_envsubst_base/standard/test (hunk 4 above)
Comment (trironkk) [reply testing/marketplace/deployer_envsubst_base/standard/test]: Done!

File: testing/marketplace/deployer_envsubst_base/standard/test (hunk 7 above)
Comment (trironkk) [reply testing/marketplace/deployer_envsubst_base/standard/test]: Single quotes were necessary in Makefile, which preprocess quotes differently.

Using single quotes here will cause the string literal $app_parameters to be passed to verify.

File: testing/marketplace/deployer_envsubst_base/standard/test (hunk 6 above)
Comment (trironkk) [reply testing/marketplace/deployer_envsubst_base/standard/test]: Done!

File: testing/marketplace/deployer_envsubst_base/standard/test (hunk 5 above)
Comment (trironkk) [reply testing/marketplace/deployer_envsubst_base/standard/test]: I think `MARKETPLACE_TOOLS_TAG` is already exported by caller.

`MARKETPLACE_TOOLS_IMAGE` is not necessary - stale relic of a broken approach.

Thanks for pointing it out!

**Pull request merged**